e2e/test/scenarios/dashboard-filters/dashboard-filters-auto-apply.cy.spec.js

```diff
@@ -113,22 +120,25 @@ describe("scenarios > dashboards > filters > auto apply", () => {
       cy.wait("@updateDashboard");
       cy.findByLabelText("Auto-apply filters").should("be.checked");
     });
-    filterWidget().within(() => {
-      cy.findByText("2 selections").should("be.visible");
-      cy.get("@cardQuery.all").should("have.length", 3);
-    });
+    filterWidget().findByText("Widget").should("be.visible");
+    getDashboardCard().findByText("Rows 1-4 of 54").should("be.visible");
+    cy.get("@cardQuery.all").should("have.length", 4);
 
-    // last applied parameter values should be used when disabling auto applying filters,
-    // even if previously there were draft parameter values
+    cy.log(
+      "last applied parameter values should be used when disabling auto applying filters, even if previously there were draft parameter values",
+    );
+    filterWidget().findByText("Widget").click();
+    popover().within(() => {
+      cy.findByText("Gadget").click();
+      cy.button("Update filter").click();
+    });
     rightSidebar().within(() => {
       cy.findByLabelText("Auto-apply filters").click();
       cy.wait("@updateDashboard");
       cy.findByLabelText("Auto-apply filters").should("not.be.checked");
     });
-    filterWidget().within(() => {
-      cy.findByText("2 selections").should("be.visible");
-      cy.get("@cardQuery.all").should("have.length", 3);
-    });
+    filterWidget().findByText("2 selections").should("be.visible");
+    cy.get("@cardQuery.all").should("have.length", 5);
```

Comment:
Move number of API calls outside of `filterWidget` since I don't think it belongs there.